### ISSUE TITLE
"@typescript-eslint/no-unused-vars" false positive on optional chaining

### ISSUE THREAD

**Issue by macrozone** [typescript-eslint/typescript-eslint#2715]
I know this has been reported alot, but was always closed with "check if you have duplicate eslint". But this is not the case, i checked yarn.lock for multiple eslint versions, restarted computer, deleted node_modules and nearly sacrificed a lamb, but no luck. It won't work.

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
       "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": ["error"]
  }
}
```

```TS
const transformColorsToOptions = (product: Product) => {
 return product?.meta?.pim?.filterColors;
}


```

reports `'product' is defined but never used. (eslint@typescript-eslint/no-unused-vars)`

**Expected Result**

should not report unused variable

**Actual Result**

does report variable

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.6.0` |
| `@typescript-eslint/parser`        | `4.4.1` |
| `TypeScript`                       | `4.0.5` |
| `ESLint`                           | `7.12.1` |
| `node`                             | `12.13.1` |

**Comment by bradzacher** [comment]
Unable to repro this against master.

Likely you have multiple versions installed.

https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md#how-do-i-check-to-see-what-versions-are-installed

**Comment by macrozone** [comment]
@bradzacher thx,

as already mentioned, i don't have multiple versions installed. 

Here is the output:

```
yarn list @typescript-eslint/eslint-plugin @typescript-eslint/parser
yarn list v1.22.4
warning Filtering by arguments is deprecated. Please use the pattern option instead.
├─ @typescript-eslint/eslint-plugin@4.6.0
└─ @typescript-eslint/parser@4.4.1
✨  Done in 0.66s.
```

**Comment by bradzacher** [comment]
If you can create an isolated repro repo - I can help you look into it further.
